internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,422 @@
+	}
+
+	defer func() { // Defer disconnect client
+		err = client.Disconnect(context.Background())
+		if err != nil {
+			log.Fatalf("Failed to disconnect client: %v", err)
+		}
+	}()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	err = client.Ping(ctx, nil)
+	if err != nil {
+		log.Fatalf("Error pinging MongoDB Analytics: %v", err)
+	}
+	log.Println("Successfully connected to MongoDB Analytics node.")
+
+	db := client.Database(expandedMetricsDB)
+
+	// Get raw data, most recent stable region, and calculate energy stats
+	findCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	patchRawData, err := findRawData(findCtx, *project, version, db.Collection(rawResultsColl))
+	if err != nil {
+		log.Fatalf("Error getting raw data: %v", err)
+	}
+
+	allEnergyStats, err := getEnergyStatsForAllBenchMarks(patchRawData, db.Collection(stableRegionsColl))
+	if err != nil {
+		log.Fatalf("Error getting energy statistics: %v", err)
+	}
+	log.Println(generatePRComment(allEnergyStats, version))
+}
+
+func findRawData(ctx context.Context, project string, version string, coll *mongo.Collection) ([]RawData, error) {
+	filter := bson.D{
+		{"info.project", project},
+		{"info.version", version},
+		{"info.variant", "perf"},
+		{"info.task_name", "perf"},
+	}
+
+	cursor, err := coll.Find(ctx, filter)
+	if err != nil {
+		log.Fatalf(
+			"Error retrieving raw data for version %q: %v",
+			version,
+			err,
+		)
+	}
+	defer func() {
+		err = cursor.Close(ctx)
+		if err != nil {
+			log.Fatalf("Error closing cursor while retrieving raw data for version %q: %v", version, err)
+		}
+	}()
+
+	log.Printf("Successfully retrieved %d docs from version %s.\n", cursor.RemainingBatchLength(), version)
+
+	var rawData []RawData
+	err = cursor.All(ctx, &rawData)
+	if err != nil {
+		log.Fatalf(
+			"Error decoding raw data from version %q: %v",
+			version,
+			err,
+		)
+	}
+
+	return rawData, err
+}
+
+// Find the most recent stable region of the mainline version for a specific test/measurement
+func findLastStableRegion(project string, testname string, measurement string, coll *mongo.Collection) (*StableRegion, error) {
+	filter := bson.D{
+		{"time_series_info.project", project},
+		{"time_series_info.variant", "perf"},
+		{"time_series_info.task", "perf"},
+		{"time_series_info.test", testname},
+		{"time_series_info.measurement", measurement},
+		{"last", true},
+		{"contexts", bson.D{{"$in", bson.A{"GoDriver perf task"}}}},
+	}
+
+	findOptions := options.FindOne().SetSort(bson.D{{"end", -1}})
+
+	findCtx, cancel := context.WithTimeout(context.Background(), 180*time.Second)
+	defer cancel()
+
+	var sr *StableRegion
+	err := coll.FindOne(findCtx, filter, findOptions).Decode(&sr)
+	if err != nil {
+		return nil, err
+	}
+	return sr, nil
+}
+
+// For a specific test and measurement
+func getEnergyStatsForOneBenchmark(rd RawData, coll *mongo.Collection) ([]*EnergyStats, error) {
```

Comment:
Suggest adding a context to the function signature, as this is a blocking function.